Source/Xamarin/Prism.DryIoc.Forms/PrismApplication.cs

```diff
@@ -128,7 +128,7 @@ protected override void ConfigureViewModelLocator()
         /// <param name="view"><see cref="Page" /> navigated to</param>
         /// <param name="type"><see cref="Type" /> to resolve</param>
         /// <param name="navigationService">Overriding instance of <see cref="INavigationService" /></param>
-        protected virtual void ResolveTypeForPage(Page view, Type type, INavigationService navigationService)
+        protected internal virtual void ResolveTypeForPage(Page view, Type type, INavigationService navigationService)
```

Comment:
I don't know... this feels a little dirty to me.  What if we just created a method responsible for creating the navigation service for the page instead of doing that logic inside of the ConfigureViewModel method directly?  Maybe something like 

```
INavigationService CreateNavigationServiceForPage(Page page)
{
    var navigationService = CreateNavigationService(); 
    ((IPageAware)navigationService).Page = page; 
    return navigationService;
}
```

Thoughts?